machinelearning/pytorch.md

```diff
@@ -0,0 +1,234 @@
+# ML inference on Graviton CPUs with PyTorch
+
+**Introduction**
+
+PyTorch is an open-source machine learning framework based on the Torch library, used for applications such as computer vision and natural language processing. It can be used across training and inference of deep neural networks. This document covers how to use PyTorch based machine learning inference on Graviton CPUs, what runtime configurations are important and how to debug any performance issues. The document also covers instructions for source builds and how to enable some of the downstream features.
+
+# How to use PyTorch on Graviton CPUs
+There are multiple levels of software package abstractions available: Python wheel (easiest option but multiple wheels to be installed), AWS DLC (Deep Learning Container, comes with all the packages installed) and the Docker hub images (additionally comes with MLPerf benchmarks). Examples of using each method are below.
+
+**Using Python wheel**
+
+PyTorch wheel, starting 1.13 release, supports OneDNN+Arm Compute Library backend for Graviton CPUs that improve the performance by ~2.5x for Resnet50 and Bert inference compared to PyTorch 1.12 wheel.
+```
+pip install torch
+
+pip install torchvision
+
+pip install torch-xla
+
+pip install torchaudio
+
+pip install torchtext
+```
+
+**AWS Graviton PyTorch DLC**
+
+AWS Graviton DLCs are PyTorch1.12 based but include all the optimizations going into PyTorch1.13, hence the performance matches to 1.13.
```

Comment:
nit: As of xx/yy the AWS Graviton DLCs are PyTorch 1.12 based but include the optimizations that were included in PyTorch 1.13 so the performance is nearly identical.

Reply:
thanks. btw, Pytorch 1.13 is not yet out but all the optimizations targeted for 1.13 are already merged.